src/main/scala/com/databricks/spark/sql/perf/mllib/data/dataGeneration.scala

```diff
@@ -103,6 +103,30 @@ object DataGenerator {
 
     (sql.createDataFrame(trainPruned), sql.createDataFrame(finalTest))
   }
+
+  def generateRandString(
+      sql: SQLContext,
+      numExamples: Long,
+      seed: Long,
+      numPartitions: Int,
+      minLen: Int,
+      maxLen: Int,
+      charSet: String): DataFrame = {
+    val rdd: RDD[String] = RandomRDDs.randomRDD(sql.sparkContext,
+      new RandStringGenerator(minLen, maxLen, charSet), numExamples, numPartitions, seed)
+    sql.createDataFrame(rdd.map(Tuple1.apply)).toDF("label")
+  }
+
+  def generateOverlappedString(
```

Comment:
Define a method for generating RDD contains random overlapped strings. Generating by converting random numbers in [0, distinctCount) into strings.
Used in `StringIndexer`.